### PR TITLE
feat: Add `imageProps` to Avatar.Image

### PR DESCRIPTION
**Pull request by kamui545**
New PR for https://github.com/callstack/react-native-paper/pull/1817

It will allow you to pass any props to the `<Image>` component inside of `Avatar.Image`.
Eg: `defaultSource`, `resizeMode`, etc...

### CODE REVIEW

File: src/components/Avatar/AvatarImage.tsx [hunk 1]
```diff
@@ -16,6 +17,10 @@ type Props = React.ComponentPropsWithRef<typeof View> & {
    * Image to display for the `Avatar`.
    */
   source: ImageSourcePropType;
+  /**
+   * Props for the `Image`
+   */
+  imageProps?: ImageProps;
```
Comment (Trancever) [inline src/components/Avatar/AvatarImage.tsx:23]: I think this API is not aligned with the rest of the lib. Can we change `source` prop to accept `ImageSourcePropType` or `({ size: number }) => React.ReactNode`? Then in the render function, we can check if `typeof source === 'function'` and call it or pass it to `Image` just like it is now.

**Review by Trancever — CHANGES_REQUESTED** [review]

File: src/components/Avatar/AvatarImage.tsx (hunk 1 above)
Comment (kamui545) [reply src/components/Avatar/AvatarImage.tsx]: Just made the changes, please have a look when you have time, thanks

**Review by Trancever — APPROVED** [review]

File: src/components/Avatar/AvatarImage.tsx (hunk 1 above)
Comment (Trancever) [reply src/components/Avatar/AvatarImage.tsx]: 👍

**Comment by aprilmintacpineda** [comment]
This is misleading, I don't see how I can pass ImageProps to Avatar.Image. The original PR contained these lines:

```diff
+     const { size = defaultSize, source, style, theme, imageProps } = this.props;
```

```diff
+ {...imageProps}
```

which will indeed allow us to pass props to image, such as `onLoad`, `onError`, etc.

**Pull request merged**